firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -241,7 +248,11 @@ String getName() {
   public Task<InstallationTokenResult> getToken(boolean forceRefresh) {
     preConditionChecks();
     Task<InstallationTokenResult> task = addGetAuthTokenListener();
-    backgroundExecutor.execute(() -> doGetAuthToken(forceRefresh));
+    if (forceRefresh) {
+      backgroundExecutor.execute(this::doGetAuthTokenForceRefresh);
+    } else {
+      backgroundExecutor.execute(this::doGetAuthTokenWithoutForceRefresh);
+    }
```

Comment:
How about we just call `#doGetAuthToken` directly (or the new name respectively).
```
backgroundExecutor.execute(() -> this::doGetAuthToken(forceRefresh));
```